### PR TITLE
Otnielvh TS.ALTER command

### PR DESCRIPTION
**Pull request by otnielvh**

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -75,6 +79,26 @@ static int parseCreateArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int ar
     return REDISMODULE_OK;
 }
 
+static int parseAlterArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
```
Comment (danni-m) [inline src/module.c]: @otnielvh why create another function instead of using `parseCreateArgs`?

File: src/module.c [hunk 2]
```diff
@@ -555,6 +579,53 @@ int TSDB_create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+    Label *newLabels;
+
+    if (parseAlterArgs(ctx, argv, argc, &retentionSecs, &maxSamplesPerChunk, &labelsCount, &newLabels) != REDISMODULE_OK) {
+        return REDISMODULE_ERR;
+    }
+
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, keyName, REDISMODULE_READ|REDISMODULE_WRITE);
+    if (RedisModule_ModuleTypeGetType(key) != SeriesType) {
+        RedisModule_CloseKey(key);
+        return RedisModule_ReplyWithError(ctx,"TSDB: key doesn't exist");
```
Comment (danni-m) [inline src/module.c]: wrong key type.

File: src/module.c [hunk 3]
```diff
@@ -555,6 +579,53 @@ int TSDB_create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+    }
+    series = RedisModule_ModuleTypeGetValue(key);
+    if (RMUtil_ArgIndex("RETENTION", argv, argc) > 0) {
+        series->retentionSecs = retentionSecs;
+    }
+
+    if (RMUtil_ArgIndex("LABELS", argv, argc) > 0) {
+        // free current labels
+        if (series->labelsCount > 0) {
```
Comment (danni-m) [inline src/module.c]: @otnielvh what about re-indexing?

File: src/tests/test_module.py [hunk 4]
```diff
@@ -178,6 +218,36 @@ def test_sanity_pipeline(self):
             actual_result = r.execute_command('TS.range', 'tester', start_ts, start_ts + samples_count)
             assert expected_result == actual_result
 
+    def test_alter_cmd(self):
```
Comment (danni-m) [inline src/tests/test_module.py]: test that reindexing works as well.

File: src/module.c (hunk 1 above)
Comment (otnielvh) [reply src/module.c]: @danni-m, we agreed TS.ALTER won't change the chunk size and therefore parseCreateArgs won't suffice. We can think of a more elegant way to parse the input, but that is beyond the scope of this PR

File: src/module.c (hunk 2 above)
Comment (otnielvh) [reply src/module.c]: Correct, thank you for catching it :)

File: src/module.c (hunk 3 above)
Comment (otnielvh) [reply src/module.c]: You're right, will add

File: src/module.c [hunk 5]
```diff
@@ -555,6 +559,59 @@ int TSDB_create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+                RedisModule_FreeString(ctx, series->labels[i].key);
+                RedisModule_FreeString(ctx, series->labels[i].value);
+            }
+            free(series->labels);
+        }
+
+        // set new newLabels
+        series->labels = newLabels;
+        series->labelsCount = labelsCount;
+        IndexMetric(ctx, keyName, series->labels, series->labelsCount);
```
Comment (danni-m) [inline src/module.c]: @otnielvh are you sure this will delete previous labels from the index?

File: src/module.c [hunk 6]
```diff
@@ -22,6 +24,8 @@
 RedisModuleType *SeriesType;
 static time_t timer;
 
+enum CMD_TYPE {CREATE, ALTER};
```
Comment (danni-m) [inline src/module.c]: I don't see any use for this enum in the code.

File: src/module.c [hunk 7]
```diff
@@ -555,6 +559,59 @@ int TSDB_create(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
...
+        }
+
+        // set new newLabels
+        series->labels = newLabels;
+        series->labelsCount = labelsCount;
+        IndexMetric(ctx, keyName, series->labels, series->labelsCount);
+    }
+
+    RedisModule_CloseKey(key);
+    RedisModule_Log(ctx, "info", "altered existing series");
```
Comment (danni-m) [inline src/module.c]: I know that its seems weird, but logging per cmd is not something we do in redis.

**Pull request merged**